server/channels/app/plugin_install.go

```diff
@@ -303,10 +303,21 @@ func extractPlugin(pluginFile io.ReadSeeker, extractDir string) (*model.Manifest
 	if err != nil {
 		return nil, "", model.NewAppError("extractPlugin", "app.plugin.filesystem.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
 	}
-
-	if len(dir) == 1 && dir[0].IsDir() {
-		extractDir = filepath.Join(extractDir, dir[0].Name())
-	}
+	filteredDir := make([]os.DirEntry, 0)
```

Comment:
Let's preallocate this to match `dir` on the assumption most of the files won't be hidden:
```suggestion
	filteredDir := make([]os.DirEntry, 0, len(dir))
```